okapi-core/src/main/java/org/folio/okapi/util/CorsHelper.java

```diff
@@ -39,20 +39,17 @@ public static void addCorsHandler(Router router, final TenantManager tenantManag
       String meth = ctx.request().getHeader(HttpHeaders.ACCESS_CONTROL_REQUEST_METHOD);
       HttpMethod method = meth != null ? HttpMethod.valueOf(meth) : ctx.request().method();
       String moduleId = ctx.request().getHeader(XOkapiHeaders.MODULE_ID);
-      tenantManager.get(tenantId)
-          .onFailure(cause -> HttpResponse.responseError(ctx, 400, cause.getMessage()))
-          .onSuccess(tenant -> {
-            ModuleCache moduleCache = tenantManager.getModuleCache(tenant);
-            List<ModuleInstance> list = moduleCache.lookup(newPath, method, moduleId);
-            for (ModuleInstance mi : list) {
-              if (mi.isHandler() && mi.getRoutingEntry().isDelegateCors()) {
-                ctx.data().put(DELEGATE_CORS, true);
-                ctx.data().put(DELEGATE_CORS_MODULE_INSTANCE, mi);
-                break;
-              }
-            }
-            ctx.next();
-          });
+      Tenant t = new Tenant();
+      t.getDescriptor().setId(tenantId);
+      List<ModuleInstance> list = tenantManager.getModuleCache(t).lookup(newPath, method, moduleId);
+      for (ModuleInstance mi : list) {
+        if (mi.isHandler() && mi.getRoutingEntry().isDelegateCors()) {
+          ctx.data().put(DELEGATE_CORS, true);
+          ctx.data().put(DELEGATE_CORS_MODULE_INSTANCE, mi);
+          break;
+        }
+      }
+      ctx.next();
     });
```

Comment:
The original code returns immediately .. and ctx is changed in the background. It's not the full handler and does not "call" anything afterwards... 

```
 //handle CORS
    CorsHelper.addCorsHandler(router, tenantManager);

    if (proxyService != null) {
      router.routeWithRegex("^/_/invoke/tenant/[^/ ]+/.*")
          .handler(proxyService::redirectProxy);
      // Note: This can not go into the InternalModule, it reads the req body,
      // and then we can not ctx.reroute(). Unless we do something trickier,
      // like a new HTTP request.
    }

``` 

The /_/invoke/tenant should be handled in ONE handler.. so this code should in the redirectProxy code (or sub thereof). I also don't like the DELEGATE_CORS_MODULE_INSTANCE.. That's not needed when handled in one place. -- finally - this code (like redirectProxy) should only be in use when proxyService != null

Reply:
> The original code returns immediately .. and ctx is changed in the background. It's not the full handler and does not "call" anything afterwards...

That makes sense. Thanks. The error message (Connection was closed to mod-login-saml) was kind misleading. While debugging it, I can see calls the right mod-login-saml url with the right headers and content. I was puzzled. :)

Reply:
>The /_/invoke/tenant should be handled in ONE handler.. so this code should in the redirectProxy code (or sub thereof). I also don't like the DELEGATE_CORS_MODULE_INSTANCE.. That's not needed when handled in one place. -- finally - this code (like redirectProxy) should only be in use when proxyService != null

This will need bigger changes due to sequence of those handlers. If I understand your other message correct (`Let's leave things as they are (mostly) for now.. `), we will not change it in this PR. Let me know otherwise.